### DISCUSSION TITLE
Keyword list - Order of the displayed keywords/tags

### DISCUSSION THREAD

**Discussion by WM-Loose** [joomla/joomla-cms#47006] [2. Cold Features]
System: 
Joomla 4.2.8 PHP8.1

Problem:
If you sort the keywords (tags) in the backend as you are used to from the menus or posts (e.g. manually according to the order), it has no effect on the order displayed in the frontend. The frontend will only sort from a-z.

This suggestion did not solve the problem: 
https://forum.joomla.org/viewtopic.php?t=970050

In the backend you can sort the list manually, but it has no influence on the display in the frontend, the selection of the manual order is not available at all (you can look under the options of the Tags component). And what you set in the component has no effect on the display of the tags in the post, the tags are always sorted from A-Z here.

If there is no bug, it would make sense to include it in an upcoming version.

**Comment by Sieger66** [comment]
> The frontend will only sort from a-z.

No, i think that is not correct: 
Standard in core is, the tags displays always unfortunatly by Tags-id number in the frontend.

**Comment by dgrammatiko** [comment]
> Standard in core is, the tags displays always unfortunatly by Tags-id number in the frontend.

Nope, the tags are displayed in the order they were inserted. I had a PR that would allow reordering the tags: https://github.com/joomla/joomla-cms/pull/19642

**Comment by brianteeman** [comment]
order they were inserted === tags id

**Comment by dgrammatiko** [comment]
> order they were inserted === tags id 

You are confused. The order is essentially the order the user typed or selected the tags:
https://github.com/joomla/joomla-cms/blob/e462044fc9a6f795c67c02c98daff5fa7d71dcba/libraries/src/Form/Field/TagField.php#L97

The tags order by id you're referencing is about how the list of available tags are displayed in the dropdown: https://github.com/joomla/joomla-cms/blob/e462044fc9a6f795c67c02c98daff5fa7d71dcba/libraries/src/Form/Field/TagField.php#L124-L262

To clear this up: the dropdown for selecting a tag is ordered by ID, the display of selected tags is by the order the user inserted/typed the tags

**Comment by brianteeman** [comment]
i am not confused at all
